src/CrandallGridWorld/TeamBug.java

```diff
@@ -76,31 +69,37 @@ public boolean sameTeam(TeamBug bug) {
 
 	// Your bug (this) kills the other bug (bugToBeKilled) if it can
 	public final void kill(TeamBug bugToBeKilled) {
-		if (canKill(bugToBeKilled) && bugToBeKilled.canBeKilled(this)) {
+		if (canKill(bugToBeKilled)) {
 			bugToBeKilled.removeSelfFromGrid();
+			turnsToDeath = 0;

```

Comment:
If you are dying why would killing someone else heal you?
eitherway should be a call to heal() for clarity